From 41462c6e730ca0e63f5fed5a517052385d980c54 Mon Sep 17 00:00:00 2001
Message-Id: <41462c6e730ca0e63f5fed5a517052385d980c54.1649663437.git.geliang.tang@suse.com>
From: Kun-Chuan Hsieh <jetswayss@gmail.com>
Date: Wed, 24 Feb 2021 05:27:52 +0000
Subject: [PATCH] tools/resolve_btfids: Fix build error with older host
 toolchains
Git-commit: 41462c6e730ca0e63f5fed5a517052385d980c54
Patch-mainline: v5.12-rc3
References: bsn#127
Modified-by-SEL: No


Older libelf.h and glibc elf.h might not yet define the ELF compression
types.

Checking and defining SHF_COMPRESSED fix the build error when compiling
with older toolchains. Also, the tool resolve_btfids is compiled with host
toolchain. The host toolchain is more likely to be older than the cross
compile toolchain.

Fixes: 51f6463aacfb ("tools/resolve_btfids: Fix sections with wrong alignment")
Signed-off-by: Kun-Chuan Hsieh <jetswayss@gmail.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Jiri Olsa <jolsa@redhat.com>
Link: https://lore.kernel.org/bpf/20210224052752.5284-1-jetswayss@gmail.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/bpf/resolve_btfids/main.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/tools/bpf/resolve_btfids/main.c
+++ b/tools/bpf/resolve_btfids/main.c
@@ -265,6 +265,11 @@ static struct btf_id *add_symbol(struct
 #define SHF_COMPRESSED (1 << 11) /* Section with compressed data. */
 #endif
 
+/* Older libelf.h and glibc elf.h might not yet define the ELF compression types. */
+#ifndef SHF_COMPRESSED
+#define SHF_COMPRESSED (1 << 11) /* Section with compressed data. */
+#endif
+
 /*
  * The data of compressed section should be aligned to 4
  * (for 32bit) or 8 (for 64 bit) bytes. The binutils ld
